src/components/input-style/input-style.ts

```diff
@@ -71,19 +86,26 @@ export class MInputStyle extends ModulVue {
         });
     }
 
-    protected created(): void {
-        setTimeout(() => {
-            this.animReady = true;
-            this.setInputWidth();
-        }, 0);
+    private calcLabelOffset(value: boolean): void {
+        if (value) {
+            let label: HTMLElement | null = this.$refs.label;
+            if (label) {
+                let labelOffset: number = label.clientHeight / 2;
+                this.labelOffset = labelOffset > CSSLABELDEFAULTMARGIN ? labelOffset + 'px' : CSSLABELDEFAULTMARGIN + 'px';
+            }
+        } else {
+            this.labelOffset = CSSLABELDEFAULTMARGIN + 'px';
+        }
     }
 
-    private get hasValue(): boolean {
-        return this.hasDefaultSlot && !this.empty;
+    public get isLabelUp(): boolean {
+        let isLabelUp: boolean = (this.hasValue || (this.isFocus && this.hasValue)) && this.hasLabel;
+        this.calcLabelOffset(isLabelUp);
```

Comment:
je ne comprends pas pourquoi on fait un calcLabelOffset dans un getter. Je pense qu'il y un problème de design ici.

Reply:
Je pense que si on enleve le calcLabelOffset dans let getter isLabelUp et qu'on fait un this.isLabelUp au lieu d'utiliser le param value dans calcLabelOffset on améliore le code. Je comprends p-ê pas quelque chose, tu me diras!